gms/gossiper.cc

```diff
@@ -940,6 +941,8 @@ void gossiper::run() {
                     }
                 }).get();
             }
+
+            _direct_fd_pinger.update_generation_number(endpoint_state_map[get_broadcast_address()].get_heart_beat_state().get_generation()).get();
```

Comment:
Doesn't generation only change during start of the service?

Reply:
I think there are other ways to increase the generation number, e.g. there's the `start_gossiping` API. Although`start_gossiping` requires an earlier `stop_gossiping` (IIUC) which stops the loop, so doing this update should be sufficient in theory at the beginning of the loop. But I wanted to stay on the safe side here.